pkg/scheduler/ugm/user_tracker_test.go

```diff
@@ -196,32 +204,59 @@ func TestSetMaxLimits(t *testing.T) {
 	usage1, err := resources.NewResourceFromConf(map[string]string{"mem": "10M", "vcore": "10"})
 	if err != nil {
 		t.Errorf("new resource create returned error or wrong resource: error %t, res %v", err, usage1)
 	}
-	result := userTracker.increaseTrackedResource(hierarchy1, TestApp1, usage1)
+	result := userTracker.increaseTrackedResource(path1, TestApp1, usage1)
 	if !result {
-		t.Fatalf("unable to increase tracked resource: queuepath %+q, app %s, res %v, error %t", hierarchy1, TestApp1, usage1, err)
+		t.Fatalf("unable to increase tracked resource: queuepath %s, app %s, res %v, error %t", path1, TestApp1, usage1, err)
 	}
 
 	userTracker.setLimits(hierarchy1, resources.Multiply(usage1, 5), 5, false, false)
 	userTracker.setLimits(hierarchy5, resources.Multiply(usage1, 10), 10, false, false)
 
-	result = userTracker.increaseTrackedResource(hierarchy1, TestApp1, usage1)
+	result = userTracker.increaseTrackedResource(path1, TestApp1, usage1)
 	if !result {
-		t.Fatalf("unable to increase tracked resource: queuepath %+q, app %s, res %v", hierarchy1, TestApp1, usage1)
+		t.Fatalf("unable to increase tracked resource: queuepath %s, app %s, res %v", path1, TestApp1, usage1)
 	}
 
-	result = userTracker.increaseTrackedResource(hierarchy1, TestApp2, usage1)
+	result = userTracker.increaseTrackedResource(path1, TestApp2, usage1)
 	if !result {
-		t.Fatalf("unable to increase tracked resource: queuepath %+q, app %s, res %v", hierarchy1, TestApp2, usage1)
+		t.Fatalf("unable to increase tracked resource: queuepath %s, app %s, res %v", path1, TestApp2, usage1)
 	}
 	userTracker.setLimits(hierarchy1, usage1, 1, false, false)
 	userTracker.setLimits(hierarchy5, usage1, 1, false, false)
 }
 
+func TestUserTrackingEvents(t *testing.T) {
+	GetUserManager()
+	user := security.UserGroup{User: "test", Groups: []string{"test"}}
+	eventSystem := mock.NewEventSystem()
+	userTracker := newUserTracker(user.User, newUGMEvents(eventSystem))
+	usage := resources.NewResourceFromMap(map[string]resources.Quantity{"cpu": 1})
+
+	groupTracker := newGroupTracker(user.User, newUGMEvents(mock.NewEventSystemDisabled()))
+	userTracker.setGroupForApp(TestApp1, groupTracker)
+	userTracker.increaseTrackedResource(path1, TestApp1, usage)
+	assert.Equal(t, 2, len(eventSystem.Events))
+	assert.Equal(t, si.EventRecord_UG_APP_LINK, eventSystem.Events[0].EventChangeDetail)
+	assert.Equal(t, si.EventRecord_UG_USER_RESOURCE, eventSystem.Events[1].EventChangeDetail)
+
+	// not removing app
+	eventSystem.Reset()
+	userTracker.decreaseTrackedResource(path1, TestApp1, usage, false)
+	assert.Equal(t, 1, len(eventSystem.Events))
+
+	// removing app
+	eventSystem.Reset()
+	userTracker.decreaseTrackedResource(path1, TestApp1, usage, true)
+	assert.Equal(t, 2, len(eventSystem.Events))
+	assert.Equal(t, si.EventRecord_UG_USER_RESOURCE, eventSystem.Events[0].EventChangeDetail)
+	assert.Equal(t, si.EventRecord_UG_APP_LINK, eventSystem.Events[1].EventChangeDetail)
+}
+
```

Comment:
Same comment like group tracker test

Reply:
All events are sent from the trackers from now on.

Reply:
Done